pkg/openshift-routes/main.go

```diff
@@ -0,0 +1,30 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package platform contains the availability of the OpenShift Routes API.
+package openshift_routes
+
+// Platform holds the auto-detected platform type.
+type OpenShiftRoutesAvailability int
```

Comment:
I don't get the purpose of this package. Is it just to define the constants?

Reply:
Alright this package seems to be inspired by the `platform`  package. I would propose to merge the `platform` package with the `autodetect`. The consumer of platform needs to import autodetect as well, I don't see a reason why these packages need to be split.

Reply:
From #1418
> The detection checks if OpenShift Routes API is installed. If true, it means we're running the Operator in OpenShift. The problem is that, for the E2E tests, we're installing the OpenShift Routes in KiND (Kubernetes). So, the E2E tests think they are running in OpenShift.

Reply:
I followed @frzifus directions from #1338.

I'll merge it

Reply:
?

Reply:
@frzifus What I wanted to say: as @pavolloffay, I followed the implementation  `platform`. Following your directions during our conversation talking about #1338.